core/federated/RTI/rti_local.c

```diff
@@ -0,0 +1,206 @@
+/**
+ * @file
+ * @author Erling Jellum (erling.r.jellum@ntnu.no)
+ * @author Edward A. Lee (eal@berkeley.edu)
+ * @author Chadlia Jerad (chadlia.jerad@ensi-uma.tn)
+ * @author Soroush Bateni (soroush@utdallas.edu)
+ * @copyright (c) 2020-2023, The University of California at Berkeley
+ * License in [BSD 2-clause](https://github.com/lf-lang/reactor-c/blob/main/LICENSE.md)
+*/
+
+#ifdef LF_ENCLAVES
+#include "rti_local.h"
+#include "rti_common.h"
+#include "util.h"
+#include "platform.h"
+#include "environment.h"
+#include "trace.h"
+#include "reactor.h"
+
+
+// The following are macros for to be used in the enter and exit
+// of the local RTI API. We extract this to macros to make sure that 
+// mutexes are acquired and released in the correct order.
+#define LTC_LOCKED_PROLOGUE(enclave) \
```

Comment:
It's not clear how this macro helps at all with making sure the order of mutex locks is correct.  This just unlocks the enclave mutex, presumably setting the stage for acquiring the global mutex?  If anything, this seems to obscure the order.  Perhaps better would be to document what the order should be.  Is the rule that you should never acquire the global mutex while holding an enclave mutex?  What about acquiring the mutex of another enclave?

Reply:
You might be right that I could also just do the locking and unlocking inside the functions and document it. For the LTC function, we only release the local enclave mutex becasue the locking of the RTI mutex is done inside the function we call next, in `rti_common.c`.